File: Array_Functions/Finding_Mex/mex.py

def mex(arr):
        n = len(arr)
        seen = [False] * (n + 1)  


        for m in arr:
            if 0 < m <= n:
                seen[m] = True

        for i in range(0, n + 1):
            if not seen[i]:
                return i

        return n + 1

if __name__ == '__main__':

	arr = [1, 0, 2, 4]

	print(mex(arr))

File: Array_Functions/Finding_Mex/mex.java

public class mex {

    public static int mex(int[] arr) {
        int n = arr.length;
        boolean[] seen = new boolean[n + 1];

        for (int m : arr) {
            if (m > 0 && m <= n) {
                seen[m] = true;
            }
        }
        for (int i = 0; i <= n; i++) {
            if (!seen[i]) {
                return i;
            }
        }
        return n + 1;
    }
	public static void main(String[] args)
{
	int[] arr = { 1, 0, 2, 4 };

	System.out.print(mex(arr));
}
}

